source/slang/slang-ir-peephole.cpp

```diff
@@ -250,6 +251,27 @@ struct PeepholeContext : InstPassBase
 
         switch (inst->getOp())
         {
+        case kIROp_AlignOf:
+            // Fold all calls to builtin functions that results in simple integer values.
+            // When postInlining flag is set, this happens for alignOf<T>() function, which
+            // infers the size of T and replaces the use with the IntLiteral.
+            if (isPostInlining && inst->getDataType()->getOp() == kIROp_IntType)
+            {
+                auto alignOfInst = as<IRAlignOf>(inst);
+                auto baseType = alignOfInst->getBaseOp()->getDataType();
+                if (as<IRParam>(baseType))
```

Comment:
Instead of doing this, we should extend `getNaturalSizeAndAlignment` to `tryGetNaturalSizeAndAlignment` that returns SLANG_FAIL if it encounters things it does not understand. And when `tryGetNaturalSizeAndAlignment` fails, we don't apply any changes here.

Reply:
i didn't do this, because  I'm not clear on how to prevent getNaturalSizeAndAlignment from crashing.

Reply:
worked with Jay on modifying this, to not explicitly check for Op param.